pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -814,13 +819,25 @@ func newAWSCloud(config io.Reader, awsServices Services) (*Cloud, error) {
 	}
 
 	selfAWSInstance, err := awsCloud.buildSelfAWSInstance()
-	if err != nil {
+	if err != nil && (cfg.Global.VPCID == "" || cfg.Global.KubernetesClusterTag == "") {
```

Comment:
I think `(cfg.Global.VPCID == "" || cfg.Global.KubernetesClusterTag == "")`, because people might have been setting KubernetesClusterTag previously?  Though admittedly none of this is widely used AFAIK.